django/contrib/staticfiles/handlers.py

```diff
@@ -1,10 +1,29 @@
+import os
+import warnings
 from urllib.parse import urlparse
 from urllib.request import url2pathname
 
+from whitenoise.base import WhiteNoise
+from whitenoise.utils import (
+    IsDirectoryError, decode_if_byte_string, ensure_leading_trailing_slash,
+)
+
 from django.conf import settings
-from django.contrib.staticfiles import utils
+from django.contrib.staticfiles import finders, utils
 from django.contrib.staticfiles.views import serve
+from django.core.exceptions import ImproperlyConfigured
 from django.core.handlers.wsgi import WSGIHandler, get_path_info
+from django.utils.deprecation import RemovedInDjango30Warning
+
+try:
+    from django.contrib.staticfiles.storage import staticfiles_storage
+except ImproperlyConfigured:
+    if not os.environ.get('DJANGO_SETTINGS_MODULE'):
```

Comment:
I think it was fixed in https://code.djangoproject.com/ticket/25034 and can be removed as long Django code used by DjangoWhiteNoise will not read settings at import time in the future.